crates/bdk/tests/wallet.rs

```diff
@@ -2188,7 +2190,7 @@ fn test_fee_amount_negative_drain_val() {
     let psbt = builder.finish().unwrap();
     let fee = check_fee!(wallet, psbt);
 
-    assert_eq!(psbt.inputs.len(), 1);
+    assert_eq!(psbt.inputs.len(), 2);
```

Comment:
This seems odd to me.
From my understanding you are not changing the PSBT inputs in this test.
Hence, why this change?
Is this test broken? Was this passing before your change?

Reply:
Since coin selection is affected by fees, likely the switch to a non-float FeeRate had some effect on the outcome of selected coins.

This will need investigation.